packages/components/src/components/drawer/drawer.jsx

```diff
@@ -15,87 +15,73 @@ const IconDrawer = ({ className }) => (
     </svg>
 );
 
-class Drawer extends React.PureComponent {
-    constructor(props) {
-        super(props);
-        this.state = { is_open: props.is_open };
-    }
+const Drawer = ({
+    anchor = 'left',
+    children,
+    className,
+    contentClassName,
+    footer,
+    header,
+    width = 250,
+    zIndex = 4,
+    ...props
+}) => {
+    const [is_open, setIsOpen] = React.useState(props.is_open);
 
-    componentWillReceiveProps(nextProps) {
-        this.setState({
-            is_open: nextProps.is_open,
-        });
-    }
+    React.useEffect(() => {
+        setIsOpen(props.is_open);
+    }, [props.is_open]);
```

Comment:
```suggestion
        setIsOpen(is_open);
    }, [is_open]);
```

Reply:
Same here, this is to avoid confusion between similar name